File: week2/algorithm_search/linear_search/linear.c
#include <stdio.h>

int main(void)
{
    int arr[] = {5, 2, 1, 3, 6, 4};
    int l = sizeof(arr) / sizeof(arr[0]);
    int target = 4;

    for (int i = 0; i < l; i++)
    {
        if (arr[i] == target)
        {
            printf("TRUE\n");
            return 0;
        }
    }

    printf("FALSE\n");
}

File: week2/algorithm_search/linear_search/linear.ts
function linear_search(arr: number[], target: number): boolean {
    for (let i = 0, l = arr.length; i < l; i++) {
      if (arr[i] === target) {
        return true;
      }
    }

    return false;
  }

  console.log(linear_search([5, 2, 1, 3, 6, 4], 8));
